metals/src/main/scala/scala/meta/metals/MetalsServices.scala

```diff
@@ -423,39 +411,33 @@ class MetalsServices(
       response
     case SbtConnect =>
       Task {
-        if (!sbtServerEnabled()) {
-          showMessage.error("Set metals.sbt.enabled=true to use sbt server.")
-        } else {
-          connectToSbtServer()
-        }
+        connectToSbtServer()
         Right(Json.Null)
       }
   }
 
-  private def sbtCompile(): Unit = sbtServer match {
-    case None => ()
-    case Some(sbt) =>
-      // TODO(olafur) support running other commands than "compile"
-      // running top-level "compile" is sub-optimal for large builds
-      // especially cross-built builds with scala.js/native
-      Sbt
-        .exec(latestConfig().sbt.command)(sbt.client)
-        .onErrorRecover {
-          case NonFatal(err) =>
-            // TODO(olafur) figure out why this "broken pipe" is not getting
-            // caught here.
-            logger.error("Failed to send sbt compile", err)
-            showMessage.warn(
-              "Lost connection to sbt server. " +
-                "Restart the sbt session and run the 'Re-connect to sbt server' command"
-            )
-        }
-        .runAsync
+  private def sbtExec(): Unit = sbtServer.foreach { sbt =>
+    // TODO(olafur) support running other commands than "compile"
```

Comment:
Outdated comment?

Reply:
I left it because of this point which is still valid:

> running top-level "compile" is sub-optimal for large builds
> especially cross-built builds with scala.js/native

But I guess, we cannot do much here. So I'll remove it.